packages/core/src/v3/runtime/prodRuntimeManager.ts

```diff
@@ -139,10 +139,14 @@ export class ProdRuntimeManager implements RuntimeManager {
       return;
     }
 
-    if (completion.ok) {
+    if (!wait.reject) {
       wait.resolve(completion);
     } else {
-      wait.reject(completion);
+      if (completion.ok) {
+        wait.resolve(completion);
+      } else {
+        wait.reject(completion);
+      }
```

Comment:
@nicktrn I don't know enough about this to review it

Reply:
Looks like `batchTriggerAndWait()` should never throw, so we always return and array of all run results, failed and successful. This is different to `triggerAndWait()` where we throw on failure. Good fix!